modules/sor/lib/poolsV3/basePoolV3.ts

```diff
@@ -145,10 +150,22 @@ export class BasePoolV3 {
                 this.hookState,
             );
         }
+
+        if (mutateBalances) {
+            tIn.increase(swapAmount.amount);
+            tOut.decrease(calculatedAmount);
+            this.poolState.balancesLiveScaled18 = this.tokens.map((t) => t.scale18);
+        }
```

Comment:
Make sure pool state is updated between swaps, so when it's reused on another swap path, its balances are correct